wireapp/kalium
# feat: avs federation calling support (AR-1680)

----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

Adds support for federated calls, passing to AVS the parsed id, this is: 
- if federated, always with domain
- if not federated, without domain

https://wearezeta.atlassian.net/wiki/spaces/ENGINEERIN/pages/555090587/Use+case+deciding+which+backend+API+version+to+use#Step-5%3A-AVS-and-API-versioning

### Solutions

Created a component `FederatedIdMapper` that will handle the parsing of the ids, depending on the config prev. stored from `/api-version` endpoint. These values are stored in kalium preferences, so we can use it without using coroutines, and reuse it in several parts of the lib

### Testing

#### Test Coverage (Optional)


#### How to Test

You should be able to call in federated environments and no federated environments. Prev. to this PR. you can't start or receive calls in federated environments, ie: anta, bella

----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
